must-gather/internal/api/logger.go

```
@@ -0,0 +1,47 @@
+}
+
+// NewLogger creates a new logger that writes to the given writer
+func NewLogger(w io.Writer) Logger {
+	return &DefaultLogger{writer: w}
+}
+
+// Log writes a timestamped log message
+func (l *DefaultLogger) Log(format string, args ...interface{}) {
+	timestamp := time.Now().Format("2006-01-02 15:04:05")
+	message := fmt.Sprintf(format, args...)
+	fmt.Fprintf(l.writer, "%s %s\n", timestamp, message)
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprintf` is not checked

(errcheck)